src/createServer.js

```
@@ -1,8 +1,338 @@
+      'userId',
+      'spentAt',
+      'title',
+      'amount',
+      'category',
+      'note',
+    ];
+
+    // Gather only provided, allowed fields
+    const updates = {};
+
+    allowedFields.forEach((field) => {
+      if (req.body[field] !== undefined) {
+        updates[field] = req.body[field];
+      }
+    });
+
+    if (Object.keys(updates).length === 0) {
+      return res.status(400).json({ error: 'No valid fields to update' });
+    }
+
+    try {
+      const [updatedCount] = await Expense.update(updates, {
+        where: { id: expenseId },
+      });
+
+      if (updatedCount === 0) {
+        return res.status(404).json({ error: 'Not found' });
+      }
+
+      const updatedExpense = await Expense.findByPk(expenseId);
+
+      res.status(200).json(updatedExpense);
+    } catch (err) {
+      res.status(500).json({ error: 'Internal Server Error' });
+    }
+  });
+
+  app.delete('/expenses/:expenseId', async (req, res) => {
+    const expenseId = req.params.expenseId;
+
+    if (!expenseId) {
+      return res.status(400).json({ error: 'Bad request' });
+    }
+
+    try {
+      const expense = await Expense.destroy({ where: { id: expenseId } });
+
+      if (!expense) {
+        return res.status(404).json({ error: 'Not found' });
+      }
+
+      res.status(204).end();
+    } catch (err) {
+      res.status(500).json({ error: 'Internal Server Error' });
+    }
+  });
+
+  app.get('/categories', async (req, res) => {
+    try {
+      const categories = await Category.findAll();
+
+      res.status(200).json(categories);
+    } catch (err) {
+      res.status(500).json({ error: 'Internal Server Error' });
+    }
+  });
+
+  app.post('/categories', async (req, res) => {
+    const { name } = req.body;
+
+    if (!name) {
+      return res.status(400).json({ error: 'Name is required' });
+    }
+
+    try {
+      const newCategory = await Category.create({ name });
+
+      res.status(201).json(newCategory);
+    } catch (err) {
```

POST /categories catch block returns 500 for all errors. Because Category.name is declared unique (see model), attempting to create a duplicate will raise a Sequelize unique-constraint error. Per requirements, duplicate creation should return a validation error (4xx). Update this catch block to detect Sequelize validation/unique errors and return 400 with a helpful message; only return 500 for unexpected server/DB connectivity errors. See Category model for the unique constraint.